parameters.go

```diff
@@ -1,16 +1,12 @@
 package winrm
 
-import (
-	"net/http"
-)
-
 // Parameters struct defines
 // metadata information and http transport config
 type Parameters struct {
 	Timeout            string
 	Locale             string
 	EnvelopeSize       int
-	TransportDecorator func(*http.Transport) http.RoundTripper
+	TransportDecorator func() Transporter
```

Comment:
This is more general indeed, but it will break users who depend on the library. It might now be possible to do it.